src/verbs/device_context.rs

```diff
@@ -142,4 +150,89 @@ impl DeviceContext {
             }
         }
     }
+
+    pub fn legacy_query_gid_table(&self) -> Result<Vec<GidEntry>, String> {
+        let mut res = Vec::new();
+        let dev_attr = self.query_device().unwrap();
+        let mut gid_type;
+
+        for port_num in 1..(dev_attr.phys_port_cnt() + 1) {
+            let port_attr = self.query_port(port_num).unwrap();
+
+            if let Some(name) = self.name() {
+                for gid_index in 0..port_attr.gid_tbl_len() {
+                    let gid = self.query_gid(port_num, gid_index).unwrap();
+                    let netdev_index;
+
+                    if gid.is_zero() {
+                        continue;
+                    }
+
+                    unsafe {
+                        gid_type = match self.query_gid_type(port_num, gid_index as u32).unwrap_unchecked() {
+                            0 if port_attr.link_layer() == IBV_LINK_LAYER_INFINIBAND => 0,
```

Comment:
Using the const defined in rdma-mummy-sys (e.g. `IBV_GID_TYPE_SYSFS_ROCE_V2`, `IBV_GID_TYPE_ROCE_V1`) is preferred over magic numbers.